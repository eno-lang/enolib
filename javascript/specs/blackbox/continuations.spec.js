const enolib = require('../..');

const input = `
Field: [value]
\\ [spaced_line_continuation]
| [direct_line_continuation]

Fieldset:
Entry = [value]
\\ [spaced_line_continuation]
| [direct_line_continuation]

List:
- [value]
\\ [spaced_line_continuation]
| [direct_line_continuation]

Empty field:
\\ [spaced_line_continuation]
| [direct_line_continuation]

Fieldset with empty entry:
Empty entry =
\\ [spaced_line_continuation]
| [direct_line_continuation]

List with empty item:
-
\\ [spaced_line_continuation]
| [direct_line_continuation]
`.trim();

describe('Blackbox test', () => {
  describe('Continuations', () => {
    it('performs as expected', () => {
      expect(enolib.parse(input).raw()).toMatchSnapshot();
    });
  });
});
