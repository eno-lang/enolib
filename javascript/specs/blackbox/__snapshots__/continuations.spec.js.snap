// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blackbox test Continuations performs as expected 1`] = `
Array [
  Object {
    "Field": "[value] [spaced_line_continuation][direct_line_continuation]",
  },
  Object {
    "Fieldset": Array [
      Object {
        "Entry": "[value] [spaced_line_continuation][direct_line_continuation]",
      },
    ],
  },
  Object {
    "List": Array [
      "[value] [spaced_line_continuation][direct_line_continuation]",
    ],
  },
  Object {
    "Empty field": "[spaced_line_continuation][direct_line_continuation]",
  },
  Object {
    "Fieldset with empty entry": Array [
      Object {
        "Empty entry": "[spaced_line_continuation][direct_line_continuation]",
      },
    ],
  },
  Object {
    "List with empty item": Array [
      "[spaced_line_continuation][direct_line_continuation]",
    ],
  },
]
`;
